irodsConnector/session.py

```diff
@@ -74,7 +108,7 @@ def default_resc(self) -> str:
             Resource name.
 
         """
-        return self.ienv.get('irods_default_resource', '')
+        return self._ienv.get('irods_default_resource', '')
 
     @property
     def host(self) -> str:
```

Comment:
There are a lot of similar property code that checks the session and passes it through if the session exists (not None). I would suggest using the special `__getattr__` method. Something like:

```python
def __getattr__(self, attr: str):
    if attr in ["host", "username", "port", ...]:
        if self.irods_session:
            return getattr(self.irods_session, attr)
        # Might want to throw an error here.
   return object.__getattribute__(self, attr)
```
I haven't tested it, but something along those lines would save a lot of work and make it easier to add new attributes to be passed through.

Reply:
Interesting approach, but the developer (and more particularly a user) will need to know a given attribute is available as its value is only returned when asked for and is never _really_ an attribute of this object.  A property is always visible and has the added benefit of custom functionality without resorting to a long list of conditionals in `__getattr__()`.  For instance:

```python
        if attr == 'attr1':
            # do something
        if attr == 'attr2':
            # do domething else
        # etc.
```

It does keep the functionality consolidated, however.  Perhaps I'll come across a use for this at some point.

Reply:
If the attributes are all different, then yes, a property for each makes much more sense.

With visibility, I assume you mean being able to autocomplete in iPython/IDE? Yes, I guess that is a disadvantage (though technically it should be possible for the IDE to resolve it). The code to me just seemed like it was passing through all the attributes from the `irods_session`.  You could also consider passing through *all* attributes from the `irods_session` object, but by that point you're probably better off just inheriting from there..